syntax = "proto3";

import "shared.proto";
import "entity.proto";

package quran;

message EditionList {
    repeated Edition edition_list = 1;
}

service EditionSvc {
    rpc CreateEdition (Edition) returns (Edition) {}
    rpc GetAll (EmptyMessage) returns (EditionList) {}
    rpc FindEditionById (IDRequest) returns (Edition) {}
    rpc FindEditionByLanguage (NameRequest) returns (EditionList) {}
    rpc FindEditionByName (NameRequest) returns (EditionList) {}
    rpc FindEditionByEnglishName (NameRequest) returns (EditionList) {}
    rpc FindEditionByType (NameRequest) returns (EditionList) {}
    rpc FindEditionByFormat (NameRequest) returns (EditionList) {}
}
